opm/simulators/flow/BlackoilModelEbos.hpp

```diff
@@ -451,13 +465,13 @@ namespace Opm {
 
                 Scalar saturationsNew[FluidSystem::numPhases] = { 0.0 };
                 Scalar oilSaturationNew = 1.0;
-                if (FluidSystem::phaseIsActive(FluidSystem::waterPhaseIdx)) {
+                if (FluidSystem::phaseIsActive(FluidSystem::waterPhaseIdx) && !has_micp_) {
                     saturationsNew[FluidSystem::waterPhaseIdx] = priVarsNew[Indices::waterSaturationIdx];
```

Comment:
Is this has_micp_ related to issues with running one-phase only water?

Reply:
Yes, I have made it more general to include all single-phase with water cases, see latest commit.